cime_config/machines/config_machines.xml

```diff
@@ -1269,23 +1269,25 @@
         <command name="load">cmake/3.21.3</command>
         <command name="load">cray-hdf5-parallel/1.12.2.1</command>
         <command name="load">cray-netcdf-hdf5parallel/4.9.0.1</command>
-	<command name="load">cray-parallel-netcdf/1.12.3.1</command>
+        <command name="load">cray-parallel-netcdf/1.12.3.1</command>
         <command name="unload">darshan-runtime</command>
       </modules>
     </module_system>
-    
+
     <RUNDIR>$CIME_OUTPUT_ROOT/$CASE/run</RUNDIR>
     <EXEROOT>$CIME_OUTPUT_ROOT/$CASE/bld</EXEROOT>
     <TEST_TPUT_TOLERANCE>0.1</TEST_TPUT_TOLERANCE>
     <MAX_GB_OLD_TEST_DATA>0</MAX_GB_OLD_TEST_DATA>
     <environment_variables>
       <env name="NETCDF_PATH">$ENV{NETCDF_DIR}</env>
       <env name="PNETCDF_PATH">$ENV{PNETCDF_DIR}</env>
+      <env name="HDF5_ROOT"/> <!-- frontier cmake cannot find_package(HDF5) for some reason, so disable it -->
       <env name="MPICH_GPU_SUPPORT_ENABLED">1</env>
       <env name="MPICH_VERSION_DISPLAY">1</env>
       <env name="MPICH_OFI_CXI_COUNTER_REPORT">2</env>
+      <env name="MPICH_CXX">$SHELL{which hipcc}</env>
```

Comment:
why is mpich connected to the version of hipcc?

Reply:
Same reason we use OMPI_CXX=nvcc_wrapper on CUDA machines. We want to use hipcc as the compiler underlying mpicxx.

Reply:
i re-read the PR and i see there was already a comment on MPICH_CXX, but i am still confused how exactly this variable is processed, if you can elaborate please? thanks.